### PR TITLE
Add Test Suite for vmware_vswitch

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fix adds test suite for vwmare_switch

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
test/integration/targets/vmware_guest_find/tasks/main.yml
test/integration/targets/vmware_vswitch/aliases
test/integration/targets/vmware_vswitch/tasks/main.yml

##### ANSIBLE VERSION
```
2.4devel
```

### CODE REVIEW

File: test/integration/targets/vmware_vswitch/aliases [hunk 1]
```diff
@@ -0,0 +1,3 @@
+posix/ci/cloud/vcenter
+cloud/vcenter
+skip/python3
```
Comment (gundalow) [inline test/integration/targets/vmware_vswitch/aliases]: What's needed to make this support Python3?
Generally speaking we shouldn't be adding `skip/python3` in, we need to fix the issues

**Comment by gundalow** [comment]
CI tests for `vmware_guest_find` are failing. Is this related to the changes in 

```
2017-06-26 04:13:55 TASK [vmware_guest_find : find folders for each vm] ****************************
2017-06-26 04:13:55 task path: /root/ansible/test/integration/targets/vmware_guest_find/tasks/main.yml:35
...
2017-06-26 04:13:58 Using module file /root/ansible/lib/ansible/modules/cloud/vmware/vmware_guest_find.py
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392 `" && echo ansible-tmp-1498450438.63-139967742216392="` echo /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392 `" ) && sleep 0'
2017-06-26 04:13:58 <127.0.0.1> PUT /tmp/tmpAqL9Qf TO /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/vmware_guest_find.py
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/ /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/vmware_guest_find.py && sleep 0'
2017-06-26 04:13:58 <127.0.0.1> EXEC /bin/sh -c '/tmp/ansible-test-coverage-ntvRjl/coverage/injector.py /root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/vmware_guest_find.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1498450438.63-139967742216392/" > /dev/null 2>&1 && sleep 0'
2017-06-26 04:13:58 The full traceback is:
2017-06-26 04:13:58 Traceback (most recent call last):
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 281, in <module>
2017-06-26 04:13:58     main()
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 267, in main
2017-06-26 04:13:58     uuid=module.params['uuid']
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 110, in getvm_folder_paths
2017-06-26 04:13:58     self.getfolders()
2017-06-26 04:13:58   File "/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py", line 191, in getfolders
2017-06-26 04:13:58     self.folders = self._build_folder_tree(self.datacenter.vmFolder)
2017-06-26 04:13:58 AttributeError: 'NoneType' object has no attribute 'vmFolder'
2017-06-26 04:13:58 
2017-06-26 04:13:58 failed: [testhost] (item=/F0/DC0/network/VM Network) => {
2017-06-26 04:13:58     "changed": false, 
2017-06-26 04:13:58     "failed": true, 
2017-06-26 04:13:58     "item": "/F0/DC0/network/VM Network", 
2017-06-26 04:13:58     "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 281, in <module>\n    main()\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 267, in main\n    uuid=module.params['uuid']\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 110, in getvm_folder_paths\n    self.getfolders()\n  File \"/tmp/ansible_IKCd2r/ansible_module_vmware_guest_find.py\", line 191, in getfolders\n    self.folders = self._build_folder_tree(self.datacenter.vmFolder)\nAttributeError: 'NoneType' object has no attribute 'vmFolder'\n", 
2017-06-26 04:13:58     "module_stdout": "", 
2017-06-26 04:13:58     "msg": "MODULE FAILURE", 
2017-06-26 04:13:58     "rc": 0
2017-06-26 04:13:58 }

```

**Comment by mattclay** [comment]
Changes to the vcenter-simulator docker container should be a separate PR. Once that PR is merged we can rebuild the docker image so it will be available for running the tests in this PR.

**Comment by jctanner** [comment]
@Akasurde this PR needs to be split out so we can merge just the flask change and then rebuild+publish the container.

needs_revision

**Comment by Akasurde** [comment]
@jctanner @mattclay I have added a separate PR for flask_controller #26125, will wait for that PR to get merged first.

**Comment by gundalow** [comment]
Docker container has been rebuild https://hub.docker.com/r/ansible/ansible/builds/blggemel7gnd6wbjsjwtdta/

**Comment by mattclay** [comment]
CI failure is due to the use of `__future__` imports in the module. The module validator currently doesn't support those.

**Comment by mattclay** [comment]
BTW: Issue https://github.com/ansible/ansible/issues/25954 is for the lack of `__future__` support.

**Comment by Akasurde** [comment]
@jctanner @gundalow @mattclay Thanks for reviews and comments.

**Pull request merged**